crates/bevy_ecs/src/schedule/state.rs

```diff
@@ -131,9 +132,15 @@ impl<S: States> Default for NextState<S> {
 }
 
 impl<S: States> NextState<S> {
-    /// Tentatively set a planned state transition to `Some(state)`.
+    /// Tentatively set a planned state transition to `state`.
     pub fn set(&mut self, state: S) {
-        self.0 = Some(state);
+        self.0 = Some((state, false));
+    }
+
+    /// Tentatively set a planned state transition to `state`,
+    /// even if transitioning from the current state to itself.
+    pub fn set_forced(&mut self, state: S) {
```

Comment:
We should consider having a `reload` method as well, which reloads the same state.

Reply:
That would make sense. It would be very simple to add that to `ChangeState` (https://github.com/bevyengine/bevy/pull/11160), whereas adding it to `NextState` would require another flag.